django/views/decorators/cache.py

```diff
@@ -48,7 +49,8 @@ def never_cache(view_func):
     """
     @wraps(view_func)
     def _wrapped_view_func(request, *args, **kwargs):
-        if not isinstance(request, HttpRequest):
+        # Don't check HttpRequest instance to allow duck-typing.
+        if not hasattr(request, 'META'):
```

Comment:
I really wanted to be smarter here, but couldn't thing of anything better. At worst, we can miss some misuse :shrug:

Reply:
I'd probably phase the comment in the positive... 

```
            # Ensure argument looks like a request.
            if not hasattr(request, 'META'):
```

I think `META` is a pretty good indicator. (The idea is to catch a moderately common mistake, not _any possible mis-use_)

Reply:
Updated.